hail/src/main/scala/is/hail/shuffler/ShuffleClient.scala

```diff
@@ -15,46 +16,101 @@ object ShuffleClient {
   private[this] val log = Logger.getLogger(getClass.getName())
 
   lazy val sslContext = {
-    val key = System.getenv("SHUFFLER_SSL_CLIENT_KEY_FILE")
-    val cert = System.getenv("SHUFFLER_SSL_CLIENT_CERT_FILE")
-    if (key == null && cert != null ||
-      cert == null && key != null) {
+    val keyStore = System.getenv("SHUFFLER_SSL_CLIENT_KEY_STORE")
+    val trustStore = System.getenv("SHUFFLER_SSL_CLIENT_TRUST_STORE")
+    if (keyStore == null && trustStore != null ||
+      trustStore == null && keyStore != null) {
       fatal("you must specify both or neither of the hail context flags: " +
-        "shuffler_ssl_client_key_file and shuffler_ssl_client_cert_file")
+        "shuffler_ssl_client_key_store and shuffler_ssl_client_trust_store")
     }
-    if (key == null) {
+    if (keyStore == null) {
       is.hail.shuffler.sslContext(
         getClass.getResourceAsStream("/non-secret-key-and-trust-stores/client-keystore.p12"),
         "hailhail",
+        "PKCS12",
         getClass.getResourceAsStream("/non-secret-key-and-trust-stores/client-truststore.p12"),
-        "hailhail"
+        "hailhail",
+        "JKS"
       )
     } else {
-      is.hail.shuffler.sslContext(key, "", cert, "")
+      is.hail.shuffler.sslContext(keyStore, "", "PKCS12", trustStore, "", "PKCS12")
+    }
+  }
+
+  def socket(): Socket = {
+    var host = System.getenv("SHUFFLER_SSL_CLIENT_HOST")
```

Comment:
I assume you're going to switch this to use deploy config at some point?  If we need another deploy config setup for deploying locally we can add that.

Reply:
Once we sort out TCP forwarding, yes, I will modify deploy config and modify this to use it.